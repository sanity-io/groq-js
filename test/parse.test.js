const {parse} = require('../src');

describe("Basic parsing", () => {
  test("Example query", () => {
    let query = `*[_type == "product"]{name}`
    let tree = parse(query);
    expect(tree).toMatchSnapshot();
  })
})

describe("Error reporting", () => {
  test("Query with syntax error", () => {
    let query = `*[_type == "]`
    try {
      parse(query);
    } catch(error) {
      expect(error.name).toBe("GroqSyntaxError")
      expect(error.position).toBe(13)
      expect(error.message).toBe("Syntax error in GROQ query at position 13")
    }
  });
})