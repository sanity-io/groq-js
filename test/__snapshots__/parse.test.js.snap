// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic parsing Example query 1`] = `
Object {
  "base": Object {
    "base": Object {
      "type": "Star",
    },
    "query": Object {
      "left": Object {
        "name": "_type",
        "type": "Identifier",
      },
      "op": "==",
      "right": Object {
        "type": "Value",
        "value": "product",
      },
      "type": "OpCall",
    },
    "type": "Filter",
  },
  "query": Object {
    "attributes": Array [
      Object {
        "key": Object {
          "type": "Value",
          "value": "name",
        },
        "type": "ObjectAttribute",
        "value": Object {
          "name": "name",
          "type": "Identifier",
        },
      },
    ],
    "type": "Object",
  },
  "type": "Projection",
}
`;
